### PR TITLE
Solr 8 11 2 upgrade

### PR DESCRIPTION
**Pull request by LeonardoGonzales**

### CODE REVIEW

File: uniprot-search/src/main/java/org/uniprot/store/search/document/suggest/SuggestDocument.java [hunk 1]
```diff
@@ -34,13 +38,22 @@ public class SuggestDocument implements Document {
...
+        public SuggestDocument build() {
+            String suggest = dictionary + "_" + id;
+            List<String> altValueList = altValues;
+            if (altValueList == null) {
+                altValueList = Collections.emptyList();
+            }
+            return new SuggestDocument(suggest, id, value, importance, altValueList, dictionary);
+        }
     }
 }
```
Comment (ahmadshadab) [inline uniprot-search/src/main/java/org/uniprot/store/search/document/suggest/SuggestDocument.java:59]: Minor - Serial version uid should change if there is any in class field

File: integration-test/src/test/java/org/uniprot/store/indexer/search/uniprot/FullCISplitNumericAnalysisSearchIT.java [hunk 2]
```diff
@@ -709,6 +709,23 @@ void canFindExactPhraseWithDash3(FullCISplitNumericAnalysisSearchIT.FieldType fi
...
+    @ParameterizedTest
+    @EnumSource(FullCISplitNumericAnalysisSearchIT.FieldType.class)
+    void canFindQueryEndingWithNumber(FullCISplitNumericAnalysisSearchIT.FieldType field) {
+        String accession = newAccession();
+
+        String indexFieldValue =
+                "Human immunodeficiency virus type 1 group M subtype B (isolate BH10) (HIV-1)";
+        String queryFieldValue = "Human immunodeficiency virus 1";
+        String query = fieldQuery(field.getQueryField(), queryFieldValue);
+
```
Comment (ahmadshadab) [inline integration-test/src/test/java/org/uniprot/store/indexer/search/uniprot/FullCISplitNumericAnalysisSearchIT.java:721]: To mimic the user's query, could we please add a search string without space between alphabet and number. e.g. Axm1

File: index-config/src/main/solr-config/uniprot-collections/publication/conf/schema.xml [hunk 3]
```diff
@@ -7,7 +7,7 @@
 	<field name="_version_" type="plong" indexed="true" stored="true" />
 
 	<!-- =============================== publication fields =============================== -->
-	<field name="id" type="string" indexed="true" stored="false"
+	<field name="id" type="string" indexed="true" stored="true"
 		required="true" multiValued="false" docValues="true" />
```
Comment (ahmadshadab) [inline index-config/src/main/solr-config/uniprot-collections/publication/conf/schema.xml:11]: What are we missing if we keep stored=false? Making this field as stored is adding more than 10GB to the publication cluster.

File: uniprot-search/src/main/java/org/uniprot/store/search/document/suggest/SuggestDocument.java [hunk 4]
```diff
@@ -34,13 +38,22 @@ public class SuggestDocument implements Document {
...
 
     // setting default field values in a builder following instructions here:
     // https://www.baeldung.com/lombok-builder-default-value
     public static class SuggestDocumentBuilder implements Serializable {
         private static final long serialVersionUID = 8082411551239368406L;
         private String importance = DEFAULT_IMPORTANCE;
+
+        public SuggestDocument build() {
+            String suggest = dictionary + "_" + id;
+            List<String> altValueList = altValues;
```
[the remark this answers is not in the document]
Comment (LeonardoGonzales) [reply uniprot-search/src/main/java/org/uniprot/store/search/document/suggest/SuggestDocument.java:52]: To remove solr warnings

File: index-config/src/main/solr-config/uniprot-collections/suggest/conf/schema.xml [hunk 5]
```diff
@@ -13,14 +13,16 @@
...
 	<field name="_version_" type="plong" indexed="true" stored="true" />
 
-	<!-- ========================== ??? uniqueKey??? =============================== -->
-
+	<!-- ========================== uniqueKey =============================== -->
+	<uniqueKey>suggest_id</uniqueKey>
 	<!-- =============================== suggestion fields =============================== -->
+	<field name="suggest_id" type="string" indexed="true"
+		stored="true" multiValued="false" />
 	<field name="id" type="basic_ci" indexed="true" stored="true"
```
[the remark this answers is not in the document]
Comment (ahmadshadab) [reply index-config/src/main/solr-config/uniprot-collections/suggest/conf/schema.xml:21]: Yes, i saw that. Can we not use id as primary/unique key?

File: index-config/src/main/solr-config/uniprot-collections/publication/conf/schema.xml (hunk 3 above)
Comment (LeonardoGonzales) [reply index-config/src/main/solr-config/uniprot-collections/publication/conf/schema.xml:11]: I removed this warning message from logs: 13:20:17.952 [coreLoadExecutor-13-thread-3] WARN  org.apache.solr.schema.IndexSchema - uniqueKey is not stored - distributed search and MoreLikeThis will not work

File: index-config/src/main/solr-config/uniprot-collections/suggest/conf/schema.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (LeonardoGonzales) [reply index-config/src/main/solr-config/uniprot-collections/suggest/conf/schema.xml:21]: Suggest collection has composite primary key (id+dict), for example, this is part of the response when we query for id:9606. (You can notice in the response that 9606 has 6 results)

`{
  "responseHeader":{
    "zkConnected":true,
    "status":0,
    "QTime":1,
    "params":{
      "q":"id:9606",
      "indent":"true",
      "q.op":"OR",
      "_forwardedCount":"1"}},
  "response":{"numFound":6,"start":0,"numFoundExact":true,"docs":[
      {
        "suggest_id":"ORGANISM_9606",
        "id":"9606",
        "value":"Homo sapiens",
        "importance":"highest",
        "altValue":["Human",
          "Man"],
        "_version_":1782811313568743442},
      {
        "suggest_id":"PROTEOME_ORGANISM_9606",
        "id":"9606",
        "value":"Homo sapiens",
        "importance":"highest",
        "altValue":["Human",
          "Man"],
        "_version_":1782811242241458196},`
        ....

**Review by ahmadshadab — APPROVED** [review]

**Pull request merged**